pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -336,6 +339,58 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 			swInfo)
 	}
 
+	// Reporting all zpools in Strorage Info
+	if vault.ReadPersistType() == types.PersistZFS {
+		zfsVersion, err := zfs.GetZfsVersion()
+		if err != nil {
+			log.Errorf("error: %v", err)
```

Comment:
Odd to proceed to report if you have this error.

Reply:
I agree here because in this case, we read the module file in sysfs and if this file is not there, then the problems will be much worse than the lack of statistics. But this should not affect the code below in any way, since it should correctly handle problems further.